File: CloudBridge-WD/src/main/java/de/r3pt1s/CloudBridge/network/protocol/packet/DisconnectPacket.java
// This class was created by r3pt1s
package de.r3pt1s.CloudBridge.network.protocol.packet;

public class DisconnectPacket extends Packet {

    public static final int SERVER_SHUTDOWN = 0;
    public static final int CLOUD_SHUTDOWN = 1;

    public Object code = 0;

    @Override
    public void encode() {
        super.encode();
        put(code);
    }

    @Override
    public void decode() {
        super.decode();
        code = get();
    }

    @Override
    public Integer getId() {
        return ID_DISCONNECT;
    }

    public static DisconnectPacket create(int code) {
        DisconnectPacket pk = new DisconnectPacket();
        pk.code = code;
        return pk;
    }
}

File: CloudBridge-PM/src/CloudBridge/network/protocol/packet/DisconnectPacket.php
<?php

namespace CloudBridge\network\protocol\packet;

class DisconnectPacket extends Packet {

    const SERVER_SHUTDOWN = 0;
    const CLOUD_SHUTDOWN = 1;

    public int $code = 0;

    public function encode(): void {
        parent::encode();
        $this->put($this->code);
    }

    public function decode(): void {
        parent::decode();
        $this->code = $this->get();
    }

    public function getId(): int {
        return self::ID_DISCONNECT;
    }

    public static function create(int $code): self {
        $pk = new self;
        $pk->code = $code;
        return $pk;
    }
}